### PR TITLE
Add diagnostics and code actions for missing case patterns

### LINKED ISSUE #520: Missing action to add missing branches in case statements using nVim + Coc
## Expected Behavior

When creating a case statement on a custom type, and only implementing one of the possible types, the code actions for the case line only display the **Add _ branch** action.

image

## Current Behavior

It should display an **Add missing branches** action, which implements placeholders for the missing values.

## Steps to Reproduce (for bugs)

1. Using nVim + Coc
2. Declare a custom type `Msg` with multiple values.
3. Implement a case statement for `Msg`, fulfilling only one branch.
4. Display code actions for the case statement line.

## Your Environment

* Version used: 2.0.2
* Editor used: nVim
* Environment name and version (e.g. node.js 5.4): v12.15.0
* Operating System and version: macOS Big Sur

### PR DESCRIPTION
**Pull request by jmbockhorst**
Fixes #520. Part of #406. This seems to work well so far. I will test this at work and add more tests for pattern matching if needed. Feel free to adjust wording of diagnostic or code action.

### CODE REVIEW

**Comment by Razzeee** [comment]
Will wait for the tests to pass before reviewing :)

**Comment by jmbockhorst** [comment]
I didn't make this compatible with node 10 since we are dropping it this version.

**Comment by Razzeee** [comment]
Should probably start with that then, so that we can rebase this. Still this also has node 14 builds failing atm.

File: src/compiler/diagnostics.ts [hunk 1]
```diff
@@ -117,6 +118,16 @@ export const Diagnostics = {
     "The `{0}` type alias cannot be followed by (..).",
     DiagnosticSeverity.Error,
   ),
+  IncompleteCasePattern: (n: number): IDiagnosticMessage =>
+    diag(
+      "incomplete_case_pattern",
+      `This \`case\` does not have branches for all possiblilities.\nMissing possibilities include:\n${getSpaces(
```
Comment (Razzeee) [inline src/compiler/diagnostics.ts:124]:
```suggestion
      `This \`case\` does not have branches for all possibilities.\nMissing possibilities include:\n${getSpaces(
```

**Pull request merged**